packages/rpc-graphql/src/loaders/__tests__/account-loader-test.ts

```diff
@@ -66,4 +66,1085 @@ describe('account loader', () => {
+                            ownerProgram {
+                                lamports
+                            }
+                        }
+                        # Mint account (see scripts/fixtures/spl-token-mint-account.json)
+                        account2: account(address: "Gh9ZwEmdLJ8DscKNTkTqPbNwLNNBjuSzaG9Vp2KGtKJr") {
+                            lamports
+                            ownerProgram {
+                                lamports
+                                ownerProgram {
+                                    lamports
+                                }
+                            }
+                        }
+                        # Stake account (see scripts/fixtures/stake-account.json)
+                        account3: account(address: "CSg2vQGbnwWdSyJpwK4i3qGfB6FebaV3xQTx4U1MbixN") {
+                            lamports
+                        }
+                        # Vote account (see scripts/fixtures/vote-account.json)
+                        account4: account(address: "4QUZQ4c7bZuJ4o4L8tYAEGnePFV27SUFEVmC7BYfsXRp") {
+                            lamports
+                            ownerProgram {
+                                space
+                            }
+                        }
+                    }
+                `;
+
+                rpcGraphQL.query(source);
+
+                // Evaluate layer 1
+                await jest.advanceTimersToNextTimerAsync();
+                expect(rpc.getMultipleAccounts).toHaveBeenCalledTimes(1);
+                expect(rpc.getMultipleAccounts).toHaveBeenLastCalledWith(
+                    [
+                        'AiZExP8mK4RxDozh4r57knvqSZgkz86HrzPAMx61XMqU',
+                        'Gh9ZwEmdLJ8DscKNTkTqPbNwLNNBjuSzaG9Vp2KGtKJr',
+                        'CSg2vQGbnwWdSyJpwK4i3qGfB6FebaV3xQTx4U1MbixN',
+                        '4QUZQ4c7bZuJ4o4L8tYAEGnePFV27SUFEVmC7BYfsXRp',
+                    ],
+                    { encoding: 'base64' }, // GraphQL client prefers `base64`
+                );
+                // Evaluate layer 2
+                await jest.advanceTimersToNextTimerAsync();
+                expect(rpc.getMultipleAccounts).toHaveBeenCalledTimes(2);
+                expect(rpc.getMultipleAccounts).toHaveBeenLastCalledWith(
+                    [
+                        '11111111111111111111111111111111',
+                        'TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA',
+                        'Vote111111111111111111111111111111111111111',
+                    ],
+                    { encoding: 'base64' }, // GraphQL client prefers `base64`
+                );
+                // Evaluate layer 3
+                await jest.advanceTimersToNextTimerAsync();
+                jest.runAllTimers();
+                expect(rpc.getAccountInfo).toHaveBeenCalledTimes(1); // Single account remaining
+                expect(rpc.getMultipleAccounts).toHaveBeenCalledTimes(2); // Hasn't been called again
```

Comment:
You can do this, but you can also ‘clear’ the mocks at each layer so that you can _actually_ write `.not.toHaveBeenCalled()` as you mean here, so that you don't have to comment the expectation to explain what's going on.

```ts
// Evaluate layer 2
/* ... */
jest.clearAllMocks();
// Evaluate layer 3
/* ... */
expect(rpc.getMultipleAccounts).not.toHaveBeenCalled();
```

This keeps the implementation, but [resets the call counts and stuff](https://jestjs.io/docs/mock-function-api#mockfnmockclear).

You can also call `rpc.getMultipleAccounts.mockClear()`.

You can also straight up reset all the mocks and prepare the `mockResolvedValue()` at every step.

Too many ways to skin this cat.